### DISCUSSION TITLE
`reset` command alters emoji rendering

### DISCUSSION THREAD

**Discussion by blieusong** [ghostty-org/ghostty#2852] [Q&A] [answered]
Hi,

I use a [Starship](https://starship.rs) prompt on a Linux container. It tells me which AWS profile is active:

![Image](https://github.com/user-attachments/assets/9b7c9b86-9b5b-4a6d-b497-f92728470581)

When I declutter my screen I use either `clear` or `reset`, whichever reach my brain faster.

Not sure why but `reset` breaks the rendering of the emoji used in the Starship prompt:

![Image](https://github.com/user-attachments/assets/3b3a922b-dfa5-46ed-a96a-c9b29bca8abd)

The cloud emoji is smaller. It does not only affect the prompt:

![Image](https://github.com/user-attachments/assets/6054bef6-1a91-4069-b486-1de1deb4d5f2)

`clear` does not have this issue.

## Setup
- Macbook Pro M2 with 32GB memory
- external monitor Samsung C49RG9x at 5120x1440
- Sequoia 15.1.1

```
$ ghostty +version
Ghostty 0.1.0-main+423f58c2

Build Config
  - Zig version: 0.13.0
  - build mode : builtin.OptimizeMode.ReleaseFast
  - app runtime: apprt.Runtime.none
  - font engine: font.main.Backend.coretext
  - renderer   : renderer.Metal
  - libxev     : main.Backend.kqueue
```

Benh

**Comment by mitchellh** [comment]
I converted this to a discussion until we can determine if this is a Ghostty bug or not. On the surface, this sounds like a starship bug. Reset allocates a new terminal and asks the shell to redraw the prompt so it shouldn’t change anything.

But it is possible. Can you record an asciinema of this and share the cast file?

**Reply by blieusong** [reply]
Hi,

I took the liberty to do that on a fresh Ubuntu image container with no Starship installed.

[ghostty.cast.zip](https://github.com/user-attachments/files/17960898/ghostty.cast.zip)

I zipped it because Github refused the *cast* file.

Here's the `test.txt` file if it helps:

[test.txt](https://github.com/user-attachments/files/17960906/test.txt)

Benh

**Reply by mitchellh** [reply, marked as answer]
Thank you very much, I see what's happening. This is indeed a bug. `reset` is resetting our grapheme clustering mode (mode 2027) back to the default. This is technically correct... but since we provide an option to start it with default on we should honor that. 

Issue: #2857

**Reply by blieusong** [reply]
Thank you for looking at it (and fixing it apparently) so quickly!
